manifests/07-downloads-deployment.yaml

```diff
@@ -59,37 +59,22 @@ spec:
         - '-c'
         - |
           cat <<EOF >>/tmp/serve.py
-          import BaseHTTPServer, os, re, signal, SimpleHTTPServer, socket, sys, tarfile, tempfile, threading, time, zipfile
+          from BaseHTTPServer import HTTPServer, BaseHTTPRequestHandler
+          from SocketServer import ThreadingMixIn
+          import SimpleHTTPServer, os, re, signal, sys, tarfile, tempfile, zipfile
+          
+          class ThreadedHTTPServer(ThreadingMixIn, HTTPServer):
```

Comment:
I added the bulk of this script in #177.  As discussed there, it was largely a copy-paste from ci-operator.  @smarterclayton added that code on 2018-06-25 in openshift/ci-operator#35.  I'm not familiar enough with its source to know what it's all about, but he links to https://stackoverflow.com/questions/46210672/ below and that says:

> No, I don't want a hack such as ThreadingMixIn that merely collects up the response and returns it as a unit.

So I expect he was aware of this approach and has consciously decided against it.  But maybe there have been stdlib improvements in 2.7 since that SO post?  I haven't hunted deeply enough to know.

Reply:
I don't understand the original motivation... The main thread accepts the client and uses a thread to handle the I/O of the client. There is not a streaming interface with the current code.